### ISSUE TITLE
Duplicate heading handling

### ISSUE THREAD

**Issue by michaelwood** [OpenDataServices/flatten-tool#345]
It would be nice if the behaviour of unflatten when coming across duplicate headings automatically treated it as an array so that you don't have to explicitly number the headers. e.g. dog/owner/0 dog/owner/1 dog/owner/2  **vs** dog/owner , dog/owner, dog/owner  . 

My actual use case is where I have some data which has some headings that I want to keep but just put in an array.

```csv
name, other_info, other_info, other_info 
bob, stuff, not interested, might be in the future
```

**Comment by jpmckinney** [comment]
Isn't there a risk of a user accidentally having a duplicate heading, in which case the tool will silently put the values into an array, rather than warn/error about the duplicate heading? In my experience with user-created spreadsheets (especially wide spreadsheets), accidental duplicates are common.

**Comment by michaelwood** [comment]
Yeah good idea, maybe a warning such as "Warning: Duplicate headings detected. Ignoring all but the first instance. Use --duplicate-headings-as-array to treat these as an array" 

I am struggling to remember my full use case now but I think it was: web form check boxes -> spreadsheet -> flatten-tool -> JSON

**Comment by jpmckinney** [comment]
I am also thinking this only works for top-level fields with literal values. If a sheet has 10 pairs of parties/id and parties/name, it could become really confusing (or hard to program) how to combine the fields into objects (imagine only the third pair has an additional parties/roles column).
